osscan2.cc

```diff
@@ -2192,7 +2195,7 @@ int HostOsScan::send_icmp_echo_probe(HostOsScanStats *hss,
   ethptr = hss->fill_eth_nfo(&eth, ethsd);
 
   for (decoy = 0; decoy < o.numdecoys; decoy++) {
-    packet = build_icmp_raw(&o.decoys[decoy], hss->target->v4hostip(),
+    packet = build_icmp_raw(&((struct sockaddr_in *)&o.decoys[decoy])->sin_addr, hss->target->v4hostip(),
```

Comment:
I haven't added any extra code for IPv6 here this all is to ensure backward compatibility with IPv4. Once approved by mentor that I am going in right direction then only I will resume the work here and possibly then we can look for changing code to a greater extant to make it more IP agnostic.